src/modules/dashboard/components/ExpenditurePage/Stages/Stages.tsx

```diff
@@ -0,0 +1,176 @@
+import React, { useState } from 'react';
+import {
+  defineMessages,
+  FormattedMessage,
+  MessageDescriptor,
+} from 'react-intl';
+import Button from '~core/Button';
+import { useDialog } from '~core/Dialog';
+import Icon from '~core/Icon';
+import StakeExpenditureDialog from './StakeExpenditureDialog';
+import StageItem from './StageItem';
+
+import styles from './Stages.css';
+
+const MSG = defineMessages({
+  stages: {
+    id: 'dashboard.Expenditures.Stages.stages',
+    defaultMessage: 'Stages',
+  },
+  notSaved: {
+    id: 'dashboard.Expenditures.Stages.notSaved',
+    defaultMessage: 'Not saved',
+  },
+  submitDraft: {
+    id: 'dashboard.Expenditures.Stages.submitDraft',
+    defaultMessage: 'Submit draft',
+  },
+  lockValues: {
+    id: 'dashboard.Expenditures.Stages.lockValues',
+    defaultMessage: 'Lock values',
+  },
+  escrowFunds: {
+    id: 'dashboard.Expenditures.Stages.escrowFunds',
+    defaultMessage: 'Escrow funds',
+  },
+  releaseFunds: {
+    id: 'dashboard.Expenditures.Stages.releaseFunds',
+    defaultMessage: 'Release funds',
+  },
+  claim: {
+    id: 'dashboard.Expenditures.Stages.claim',
+    defaultMessage: 'Claim',
+  },
+  completed: {
+    id: 'dashboard.Expenditures.Stages.completed',
+    defaultMessage: 'Completed',
+  },
+  draft: {
+    id: 'dashboard.Expenditures.Stages.draft',
+    defaultMessage: 'Draft',
+  },
+  locked: {
+    id: 'dashboard.Expenditures.Stages.locked',
+    defaultMessage: 'Locked',
+  },
+  funded: {
+    id: 'dashboard.Expenditures.Stages.funded',
+    defaultMessage: 'Funded',
+  },
+  released: {
+    id: 'dashboard.Expenditures.Stages.released',
+    defaultMessage: 'Released',
+  },
+  claimed: {
+    id: 'dashboard.Expenditures.Stages.claimed',
+    defaultMessage: 'Claimed',
+  },
+});
+
+interface ActiveState {
+  id: number;
+  label: string | MessageDescriptor;
+  buttonText: string | MessageDescriptor;
+  buttonAction: () => void;
+}
+
+const Stages = () => {
+  const [activeState, setActiveState] = useState<ActiveState | null>(null);
+
+  const openDraftConfirmDialog = useDialog(StakeExpenditureDialog);
+
+  const states = [
+    {
+      id: 1,
```

Comment:
I think we should refactor the `id`s to actually be `enums` as it will make them easier to read in code